File: services/backend/src/core_service/modules/notifications/domain/notification_setting.py
"""NotificationSetting 도메인 엔티티 — schema.md §3.15 `notification_settings` 매핑.

WF5 화면의 "수신 대상·채널 선택" — **이미 발생한 알림을 누가·어떤 채널로 받을지**만
다룬다. 알림을 보낼지 말지의 임계치 로직(정서 모니터링 등)은 이 범위 밖이며
decisions.md #12(법무검토 대기)와 충돌하지 않도록 의도적으로 제한됨.

한 family_member는 채널(sms/email/push)마다 최대 1개 설정을 가진다
(마이그레이션 0005의 `uq_notification_settings_member_channel`).
"""

from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum
from uuid import UUID


class NotifyChannel(StrEnum):
    SMS = "sms"
    EMAIL = "email"
    PUSH = "push"


@dataclass
class NotificationSetting:
    id: UUID
    family_member_id: UUID
    channel: NotifyChannel
    receives_emotion_alerts: bool
    receives_chapter_updates: bool
    receives_sync_issues: bool
    updated_at: datetime

File: apps/web/src/types/notification-setting.ts
/** design.md §4.2 / schema.md `notification_settings` — WF5 알림 수신 설정.
 * PUT은 이 구성원의 설정 전체를 통째로 교체(delete→insert)한다. */

export type NotifyChannel = "sms" | "email" | "push";

export interface NotificationSetting {
  id: string;
  familyMemberId: string;
  channel: NotifyChannel;
  receivesEmotionAlerts: boolean;
  receivesChapterUpdates: boolean;
  receivesSyncIssues: boolean;
  updatedAt: string;
}

/** PUT 요청 1건 — 한 채널의 수신 항목 스위치. */
export interface ChannelPreference {
  channel: NotifyChannel;
  receivesEmotionAlerts: boolean;
  receivesChapterUpdates: boolean;
  receivesSyncIssues: boolean;
}

export const NOTIFY_CHANNELS: readonly NotifyChannel[] = ["push", "email", "sms"];

export const NOTIFY_CHANNEL_LABEL: Record<NotifyChannel, string> = {
  sms: "문자(SMS)",
  email: "이메일",
  push: "앱 푸시",
};

/** 채널당 켜고 끄는 boolean 수신 항목 키. */
export type NotifyItemKey = Exclude<keyof ChannelPreference, "channel">;

/** 수신 항목 3종. `receivesEmotionAlerts`는 CTO 검토 B1에 따라 기본 opt-out(false). */
export const NOTIFY_ITEMS = [
  { key: "receivesChapterUpdates", label: "자서전 챕터 갱신" },
  { key: "receivesSyncIssues", label: "동기화 문제" },
  { key: "receivesEmotionAlerts", label: "정서 이상 알림" },
] as const satisfies readonly { key: NotifyItemKey; label: string }[];
